python/ray/tests/test_streaming_generator.py

```diff
@@ -799,18 +812,170 @@ async def main():
     print(summary)
 
 
-def test_reconstruction(ray_start_cluster):
-    cluster = ray_start_cluster
-    # Head node with no resources.
-    cluster.add_node(
-        num_cpus=0,
-        _system_config=RECONSTRUCTION_CONFIG,
-        enable_object_reconstruction=True,
-    )
-    ray.init(address=cluster.address)
-    # Node to place the initial object.
-    node_to_kill = cluster.add_node(num_cpus=1, object_store_memory=10**8)
-    cluster.wait_for_nodes()
+def test_generator_wait(shutdown_only):
+    """
+    Make sure the generator works with ray.wait.
+    """
+    ray.init(num_cpus=8)
+
+    @ray.remote
+    def f(sleep_time):
+        for i in range(2):
+            time.sleep(sleep_time)
+            yield i
+
+    @ray.remote
+    def g(sleep_time):
+        time.sleep(sleep_time)
+        return 10
+
+    gen = f.options(num_returns="streaming").remote(1)
```

Comment:
question: how does streaming work when a task both yields and returns values?

Reply:
I think it will be the same as normal Python - `return` just raises StopIteration.

Reply:
Added a test